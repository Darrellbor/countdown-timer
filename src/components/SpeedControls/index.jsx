import React from 'react'

const SpeedControls = () => {
    return (
        <div>
            
        </div>
    )
}

export default SpeedControls;