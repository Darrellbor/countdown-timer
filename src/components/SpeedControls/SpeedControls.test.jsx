import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";

import SpeedControls from "./";

describe("Speed Controls Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SpeedControls
        theme="Dark"
        active={true}
        controlClicked={false}
        onClick={() => {}}
      >
        1X
      </SpeedControls>
    );
  });

  test("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("calls prop onClick when clicked", () => {
    const spy = sinon.spy();
    wrapper = shallow(<SpeedControls onClick={spy} />);

    wrapper.find(".SpeedControls").simulate("click");

    expect(spy.calledOnce).toBe(true);
  });

  test("renders children properly", () => {
    expect(wrapper.find(".SpeedControls__inner").contains("1X")).toBeTruthy();
  });

  test("check if theme class is applied", () => {
    expect(wrapper.exists(".SpeedControls--dark")).toBeTruthy();
  });

  test("check if active class is applied", () => {
    expect(wrapper.exists(".SpeedControls__active")).toBeTruthy();
  });
});
