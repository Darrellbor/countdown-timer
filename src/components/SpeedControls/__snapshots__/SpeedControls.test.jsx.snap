// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Speed Controls Component renders correctly 1`] = `
<div
  className="SpeedControls SpeedControls--dark false  SpeedControls__active "
  onClick={[Function]}
>
  <div
    className="SpeedControls__inner"
  >
    1X
  </div>
</div>
`;
